coralproject/talk
# [CORL-2779] create a worker based word list processor

## What does this PR do?

Replaces the existing vm based word list processor with a node worker based word list processor.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

None

## Does this PR introduce any new environment variables or feature flags?

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indices.

## How do I test this PR?

- start up Coral
- add some suspect and banned words
- post some comments with a non-staff user using those words
- check your rejected, reported, and unmoderated queues for the comments + highlighted words
- update the suspect/banned with new words
- use the new words in comments with a non-staff user
- check your rejected, reported, and unmoderated queues for the comments + highlighted words

## Where any tests migrated to React Testing Library?

None

## How do we deploy this PR?

This will cause pods to run multiple workers for word list processing, it might be advisable to consider slightly upping pod CPU resources so they can better multi-task the workers. This will effectively make the word list processing multi-threaded and allow Coral to better utilize multi-threading instead of linearly scaling on one processor as with most traditional node apps tend to be.
